app/workers/question_worker.rb

```diff
@@ -0,0 +1,37 @@
+class QuestionWorker
+  include Sidekiq::Worker
+  def perform
+    require 'post_to_slack'
+
+    # First check there is no active question already
+    if Question.where(open: true).none?
```

Comment:
This is probably fine. But it's probably best if you create a scope of this. Check out [the docs](https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope) on scope. In your `Question` model, you'll want to add something like:

```ruby
class Question < ActiveRecord::Based
  scope :open, -> { where(open: true) }
end
```

And then you can do this in your worker:

```ruby
if Question.open.none?
...
end
```

Reply:
The reason you'd create a scope is because you might check to see if there are open questions somewhere else in the code, and in the case that your criteria for "open" changes, you only have to change it in one place, your model.